prow/pjutil/pjutil.go

```diff
@@ -271,23 +272,38 @@ func ProwJobFields(pj *prowapi.ProwJob) logrus.Fields {
 // JobURL returns the expected URL for ProwJobStatus.
 //
 // TODO(fejta): consider moving default JobURLTemplate and JobURLPrefix out of plank
-func JobURL(plank config.Plank, pj prowapi.ProwJob, log *logrus.Entry) string {
+func JobURL(plank config.Plank, pj prowapi.ProwJob, log *logrus.Entry) (string, error) {
 	if pj.Spec.DecorationConfig != nil && plank.GetJobURLPrefix(pj.Spec.Refs) != "" {
 		spec := downwardapi.NewJobSpec(pj.Spec, pj.Status.BuildID, pj.Name)
 		gcsConfig := pj.Spec.DecorationConfig.GCSConfiguration
 		_, gcsPath, _ := gcsupload.PathsForJob(gcsConfig, &spec, "")
 
 		prefix, _ := url.Parse(plank.GetJobURLPrefix(pj.Spec.Refs))
-		prefix.Path = path.Join(prefix.Path, gcsConfig.Bucket, gcsPath)
-		return prefix.String()
+
+		// extract bucket name from gcsConfig
+		var bucketName string
+		bucketSplit := strings.Split(gcsConfig.Bucket, "://")
+		switch len(bucketSplit) {
+		case 1:
+			bucketName = bucketSplit[0]
+		case 2:
+			bucketName = bucketSplit[1]
+		default:
+			return "", fmt.Errorf("calculating joburl: bucket field has invalid format %q, expected either <bucket-name> or <storage-provider>://<bucket-name>", gcsConfig.Bucket)
```

Comment:
Nit: `bucket field %q has invalid format` rather than `bucket field has invalid format %q`

Reply:
Also, I think we should extract the part here that can error into a function that is run by checkconfig for all possible `DecorationConfigs`, to avoid getting to the point of deploying an invalid config

Reply:
Sounds reasonable. So implement the validation  in `checkConfig` and assume here that the error cannot happen (ignore it)?

Reply:
Implement it in pjutil, export it, call it in `checkconfig` and still return the error here. Ppl might not use checkconfig, we might implement the validation wrong or incomplete, there is not a guarantee this is correct just because we add a validation

Reply:
Couldn't add it in pjutil or the config package because of circular dependencies. I'll add it in `prow/apis/prowjobs/v1/types.go`. I don't really have an overview, but if there's a more appropriate place I can move it there.